Assets/Scripts/FacePunch/FacePunchSteamManager.cs

```diff
@@ -0,0 +1,181 @@
+* furnished to do so, subject to the following conditions:
+* 
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+* 
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+/*
+* This script is made using Facepunch.Steamworks Version 2.3.2
+*/
+
+
+#if ENABLE_FACEPUNCH_STEAM && (UNITY_STANDALONE_WIN || UNITY_STANDALONE_LINUX || UNITY_STANDALONE_OSX || STEAMWORKS_WIN || STEAMWORKS_LIN_OSX)
+
+using UnityEngine;
+
+namespace PlayEveryWare.EpicOnlineServices.Samples.Steam
+{
+    public class SteamManager : MonoBehaviour
+    {
+        protected static SteamManager s_instance;
+        public static SteamManager Instance
+        {
+            get
+            {
+                if (s_instance == null)
+                {
+                    return new GameObject("FacepunchSteamManager").AddComponent<SteamManager>();
+                }
+                else
+                {
+                    return s_instance;
+                }
+            }
+        }
+
+        protected bool m_bInitialized = false;
+        public static bool Initialized
+        {
+            get
+            {
+                return Instance.m_bInitialized;
+            }
+        }
+
+#if UNITY_2019_3_OR_NEWER
+        // In case of disabled Domain Reload, reset static members before entering Play Mode.
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        private static void InitOnPlayMode()
+        {
+            s_instance = null;
+        }
+#endif
+        protected virtual void Awake()
+        {
+            // Only one instance of DiscordManager at a time!
+            if (s_instance != null)
+            {
+                Destroy(gameObject);
+                return;
+            }
+            s_instance = this;
+
+            // We want our DiscordManager Instance to persist across scenes.
+            DontDestroyOnLoad(gameObject);
+
+            try
+            {
+                Steamworks.SteamClient.Init(1679310);
+                Debug.LogWarning("[Facepunch Steamworks] Initialized");
+            }
+            catch (System.Exception e)
+            {
+                Debug.LogError(e.Message);
+                Debug.LogError("[Facepunch Steamworks] SteamAPI_Init() failed. Refer to Valve's documentation or the comment above this line for more information.", this);
+               
+                return;
+                // Something went wrong - it's one of these:
```

Comment:
Is this comment supposed to be before the LogError? The LogError line on 90 seems to make reference to it.